driver/csiplugin/controllerserver.go

```diff
@@ -564,6 +565,46 @@ func (cs *ScaleControllerServer) CreateVolume(ctx context.Context, req *csi.Crea
 
 	glog.Infof("volume:[%v] -  spectrum scale volume create params : %v\n", scaleVol.VolName, scaleVol)
 
+	if scaleVol.IsFilesetBased && scaleVol.Compression != "" {
+		glog.Infof("createvolume: compression is enabled: changing volume name")
+		scaleVol.VolName = fmt.Sprintf("%s-COMPRESS%s", scaleVol.VolName, strings.ToUpper(scaleVol.Compression))
```

Comment:
@drolsonibm are we missing to create policy on scale for compression here like replication in below code ?

Reply:
For compression we plan to have that as a separate policy file that will have to run. The reason being is that it is not a placement rule so even with the rule in place, a cronjob or some other mechanism will still need to run the policy from time to time. We wanted to avoid running the installed policy for the filesystem since other rules may be installed that can take a long time, like migrate/recall, that we don't necessarily want to be running on their behalf. 

The separate policy file will use the mmapplypolicy <fs> -P <file> to just run the compression rules.